caches/clan.go

```diff
@@ -0,0 +1,105 @@
+package caches
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	gocache "github.com/patrickmn/go-cache"
+	"github.com/topfreegames/khan/models"
+)
+
+// ClansSummaries represents a cache for the RetrieveClansSummaries operation.
+type ClansSummaries struct {
+	// Cache points to an instance of gocache.Cache used as the backend cache object.
+	Cache *gocache.Cache
+
+	// The cache for one clan will live by a random amount of time within [TTL - TTLRandomError, TTL + TTLRandomError].
```

Comment:
Why are you using a random TTL approach?

Reply:
I don't want the load to come in spikes: When I need a DB query to retrieve clans data, I don't want it to be a huge number of clans.

Reply:
Won't requests naturally "misalign" due to players being in different places/latencies/states/etc? I'm not completely convinced the added logic is necessary either.

Regardless of that, we don't need both TTL and TTLRandomError to be supplied - we could calculate a random value using just the first (eg, `TTL/2 <= x <= TTL `).

Reply:
I agree that TTLRandomError is too much (I've just removed it), but every game probably has a global ranking of clans that won't change very much within minutes. If the randomness is not added, the SELECT query is going to retrieve all those clans at once every time. I prefer the query being executed more often but retrieve less data. On average over a time span it will be the same with either approaches, but with randomness we avoid big spikes.

Reply:
I am not bought on the need (and the complexity) regarding the random ttl. I don't see an issue of the SELECT query retrieving all clans at once, specially because it already is how it works today and, IMO, it makes it easier to reason about the data that is being cached.
How big do you expect the big spikes to be? Why do you believe this is a big concern?

Reply:
I agree it's not a game changer in this case, because the longest query I found in Jaeger took 25ms. But since I've removed the extra config parameter to implement this approach, now the choice is between

```go
ttl := cTTL + time.Duration(rand.Int63n(int64(cTTL)))
```
and
```go
ttl := cTTL
```
In this case I don't see harm in picking the first option.

Reply:
IMO the harm is making it more complex than necessary, but I won't block you from merging this because of the random TTL. 

Every time you add some code, I believe the reasoning should be to choose the easier/less complex solution unless there's an explicit need for the extra complexity. We should always ask ourselves "why should I do this?" rather than "why shouldn't I do this?". 

The complexity that I'm referring to is not only about code complexity because in this case they are pretty equivalent, but also regarding to product requirements. You won't be able to say that  the clan data become stale after 5m, for example.

We can further discuss it in person if you'd like to.

Reply:
I think you're right, I'm removing the randomness.